mempool/src/tests/node.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) The Diem Core Contributors
```

Comment:
this kinda looks like something that would make sense to generalize for StateSync too

Reply:
I agree, I'll see what I can do, but I might not get it all separated out enough to be fully reused.

Reply:
Yeah, this is a nice trick and I wondered this too.. Although, from my memory, I'm not sure how much it would buy/save us